src/azure/azureController.ts

```diff
@@ -83,4 +98,98 @@ export class AzureController {
         this.azureMessageDisplayer = new AzureMessageDisplayer();
     }
 
+    public async getTokens(profile: ConnectionProfile, accountStore: AccountStore, settings: AADResource): Promise<ConnectionProfile> {
+        let account: IAccount;
+        let config = vscode.workspace.getConfiguration('mssql').get('azureActiveDirectory');
+        if (config === utils.azureAuthTypeToString(AzureAuthType.AuthCodeGrant)) {
+            let azureCodeGrant = await this.createAuthCodeGrant();
+            account = await azureCodeGrant.startLogin();
+            await accountStore.addAccount(account);
+            const token = await azureCodeGrant.getAccountSecurityToken(
+                account, azureCodeGrant.getHomeTenant(account).id, settings
+            );
+            profile.azureAccountToken = token.token;
+            profile.email = account.displayInfo.email;
+            profile.accountId = account.key.id;
+        } else if (config === utils.azureAuthTypeToString(AzureAuthType.DeviceCode)) {
+            let azureDeviceCode = await this.createDeviceCode();
+            account = await azureDeviceCode.startLogin();
+            await accountStore.addAccount(account);
+            const token = await azureDeviceCode.getAccountSecurityToken(
+                account, azureDeviceCode.getHomeTenant(account).id, settings
+            );
+            profile.azureAccountToken = token.token;
+            profile.email = account.displayInfo.email;
+            profile.accountId = account.key.id;
+        }
+        return profile;
+    }
+
+    public async refreshTokenWrapper(profile, accountStore, accountAnswer, settings: AADResource): Promise<ConnectionProfile> {
+        let account = accountStore.getAccount(accountAnswer.key.id);
+        if (!account) {
+            throw new Error('account not found');
```

Comment:
Shouldn't this error be shown to the user? Or is it shown after this error is thrown?